Common/src/CConfig.cpp

```diff
@@ -3156,6 +3156,28 @@ void CConfig::SetPostprocessing(unsigned short val_software, unsigned short val_
     VolumeOutputFiles[2] = SURFACE_PARAVIEW_BINARY;
   }
 
+  if(OptionIsSet("DISCRETE_ADJOINT")){
+    // Check, if sensitivity is set.
+    bool sensIsSet = false;
+    for (unsigned long i  = 0; i < nVolumeOutput; i++) {
+      if(GetVolumeOutput_Field(i).compare(string("SENSITIVITY")) == 0){
+        sensIsSet = true;
+      }
+    }
+    if(!sensIsSet){
+      // Add sensitivity volume output for adjoint runs
+      string * vals = new string[nVolumeOutput+1];
+      for (unsigned long i  = 0; i < nVolumeOutput; i++) {
+        vals[i].assign(VolumeOutput[i]);
+      }
+      vals[nVolumeOutput] = string("SENSITIVITY");
+      nVolumeOutput++; //Array has one entry more now
+
+      delete [] VolumeOutput;
+      VolumeOutput = &vals[0];
```

Comment:
Is this kind of "manual push_back" done somewhere else in config?

Reply:
I'm not quite sure, what you want to say by this. 
Is it bad practise? If yes, how can I improve?

Reply:
The bad practice was making `VolumeOutput` a raw array of strings instead of `vector<string>` but that is too ingrained now.
I just wanted to make sure this snipped of code was not being duplicated, we are bound to be doing something similar in CConfig...
But actually would it not be better to move the logic of always making sure the sensitivities are added to the output classes?
You can just check `requestedVolumeFields` and if "SENSITIVITY" is not there you do a simple `requestedVolumeFields.emplace_back("SENSITIVITY")`

Reply:
I tried this idea. However, in a script run via `discrete_adjoint.py`, where `SU2_DOT` is called after `SU2_CFD_AD`, we get again the above error message, since in the file `config_DOT_AD` sensitivity is still not an volumeoutput.

Reply:
Maybe we are not talking about the same thing.
If in the output classes you add a field it will be outputted... SU2_DOT just looks for the name of the variable in the file, it does not care about the options you have in the CConfig, to be clear in CAdjFlowOutput.cpp try adding after line 65:
```
if (find(requestedVolumeFields.begin(), requestedVolumeFields.end(), string("SENSITIVITY")) == requestedVolumeFields.end()) {
  requestedVolumeFields.emplace_back("SENSITIVITY");
}
```

Reply:
Okay, I think I got your idea now. 
Pushed it :)